docs/practices/protocol_upgrade.md

```diff
@@ -23,6 +23,22 @@ especially if the change is invasive. For those changes, we may want to have
 several months of testing before we are confident that the change itself works
 and that it doesn't break other parts of the system.
 
+### Protocol version voting and upgrade
+
+When a new neard version, containing a new protocol version, is released, all node maintainers need 
+to upgrade their binary. That typically means stopping neard, downloading or compiling the new neard
+binary and restarting neard. However the protocol version of the whole network is not immediately 
+bumped to the new protocol version. Instead a process called voting takes place and determines if and 
+when the protocol version upgrade will take place. 
+
+Voting is a fully automated process in which all nodes across network (TODO - all or just validators? equal or weighted voting?)
```

Comment:
IIUC only block producers. The voting power is proportional to their stake.
The mechanism for voting is a field `latest_protocol_version` in `BlockHeader` https://github.com/near/nearcore/blob/81e92c41380ee0d47a8991bea5281295932199b4/core/primitives/src/block_header.rs#L151 where a block producer sets that it can "speak the language of protocol version YY", even though currently it "speaks the language of protocol version XX".
https://github.com/near/nearcore/blob/2b6f9f10a3afedc6f87e3170e3328104a523c0bf/chain/epoch-manager/src/lib.rs#L555